### DISCUSSION TITLE
Terraform images not updated

### DISCUSSION THREAD

**Discussion by bodinsamuel** [renovatebot/renovate#13405] [Request Help] [answered]
Hello,
 
Using renovate (not self hosted).
Managers: 
  - Terraform
  - Kubernetes (.yaml)

I don't see any update to `image` props in _.tf_ files.
It works well for `required_providers` and lockfile but does not update _.tf_ files when an image is specified.

Maybe it's too specific and not supported by renovate or there is something to do that is not clear to me.

Example I have this:
```hcl
resource "kubernetes_deployment" "datadog_agent" {
  [...]
        container {
          name  = "datadog-agent"
          image = "gcr.io/datadoghq/agent:7.31.1"
       }
  [...]
}
```

I expected to have a PR for this image. Can you provide some guidance please? ☺️

**Comment by rarkins** [comment, marked as answer]
I don't think our terraform extract function detects these types of dependencies yet. Terraform is pretty flexible so we can't automatically support any type of custom dependency defined within, even if the dependency type is already known to us (e.g. Docker image in this case).

Is `container.[name,image]` a standardized object which multiple resources use the same way, or is it something specific to `kubernetes_deployment`? If the latter, then this means the feature request should be "support kubernetes_deployment.container Docker images in terraform manager".

**Reply by viceice** [reply]
That object is specific to kubernetes.

But can be used in different types, like the deployment, pod, job, cronjob, ...

**Reply by rarkins** [reply]
In that case this needs a feature request, a definitive spec for this resource, and reproduction repo with all examples needing support

**Reply by bodinsamuel** [reply]
indeed there are others resources, I'll try to have an exhaustive list and make a feature request. Thanks 🙏🏻

**Reply by bodinsamuel** [reply]
Done here: https://github.com/renovatebot/renovate/issues/13461
